### ISSUE TITLE
Improve Voiceover support

### ISSUE THREAD

**Issue by tyrylu** [blinksh/blink#171]
Hello.
Currently, if you try to use blink with Voiceover, there are some issues:
1. The keyboard feedback is for some reasons disabled, e. g. no announcements of deleted letters, completed words etc.
2. You can not use Voiceover rotor commands to browse the text field by lines etc., you have to hear the entire (or visible?) content.
You are almost there, connecting to a ssh host with Voiceover is doable, but for some serious work, it at least needs the text control reading fixed, the keyboard can be next.
Experienced on Ios 10.1 on an Iphone 6 with the latest app store release.
I would much like to help further, but unfortunately i have no Mac os development machine.

**Comment by carloscabanero** [comment]
Thanks Tyrylu, reading it line by line definitely is the next step to take. For some reason that is working on the desktop but not on the iPad, I will add this to the next iteration. :)

**Comment by tyrylu** [comment]
Hello,
it's good that you could reproduce it and will fix it.

**Comment by mikeiit** [comment]
Hey guys,

This is Mike. Hope you guys had a good day so far. I saw this project here and I am interested to know more about it and see if I can possibly contribute to it. 

I'm an entry level developer in a startup here in Chicago. I have done some development as well as testing. Please let me know if we can have a talk next week and see if there is a chance for us to work on the project together. Thank you. 

Mike

**Comment by carloscabanero** [comment]
Leaving this here for reference: https://developer.apple.com/library/content/technotes/TestingAccessibilityOfiOSApps/TestAccessibilityiniOSSimulatorwithAccessibilityInspector/TestAccessibilityiniOSSimulatorwithAccessibilityInspector.html

**Comment by carloscabanero** [comment]
Hi Mike! Thanks for your interest in Blink! Would be great to see ways we can collaborate. Please send me a DM @BlinkShell on Twitter to discuss. :)

**Comment by yury** [comment]
We improved voice over support with our new smart keys keyboard. 
@mikeiit, @tyrylu we pushed public Test Flight version. Is possible you to try it out?

**Comment by tyrylu** [comment]
Definitely, if you can send the invite to lukastyrychtr@gmail.com.

**Comment by yury** [comment]
@tyrylu sent you invite. One note: It requires iOS/iPadOS 13

**Comment by tyrylu** [comment]
Ach... That might be a problem, i'll have to wait until i get something newer than my IPhone 6 then... :-)

**Comment by tyrylu** [comment]
I finally got a chance to test the latest beta. The support is quite good, however there are some issues.
* The Voiceover braille input does not work for some reason (it appears as if it would like to move through some webpage elements list or something like that)
* The Input field loses VoiceOver focus
* New output from the remote side does not show until the input text field is reactivated
* It would be really useful if the new output could be spoken by Vo

**Comment by yury** [comment]
It should spoke output, I will look. Thank you!

**Comment by yury** [comment]
Quick question. Did you have VO turned on before starting blink or creating new shell?

**Comment by tyrylu** [comment]
Yes, of course.

**Comment by demhademha** [comment]
Is there an update on this issue

**Comment by brandon-fryslie** [comment]
Are the maintainers still open to contributions on this?  Would love to be able to be able to dictate to Claude and get an answer back.  Using Tmux, Tailscale and voice control it's 90% of the way there.

**Comment by carloscabanero** [comment]
Of course, anything that can improve input (or anything else) in Blink is very welcome.
